### PR TITLE
fix #44328: method validation for opaque closures

### LINKED ISSUE #44328: OpaqueClosure precompilation test generates warnings in debug builds
In a debug build, `make debug test-precompile` generates the following warnings:
```
WARNING: Encountered invalid lowered code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
WARNING: Encountered invalid inferred code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
WARNING: Encountered invalid optimized code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
WARNING: Encountered invalid lowered code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
WARNING: Encountered invalid inferred code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
WARNING: Encountered invalid optimized code for method Tuple: Core.Compiler.InvalidCodeError(kind="method signature does not match number of method arguments", meta=(false, 1, 2))
```
This comes from https://github.com/JuliaLang/julia/blob/df81bf9a96c59f257a01307cd0ecf05035d8301f/test/precompile.jl#L1222
in the test harness that uses `@opaque`.

### PR DESCRIPTION
**Pull request by simeonschaub**
I believe it's intentional that for these methods, the `sig` field is
just ignored and always set to `Tuple`. Also fixes a lowering bug I
discovered that would cause errors if `Union` was shadowed.

I have verified that this fixes the reported warnings.

### CODE REVIEW

**Comment by timholy** [comment]
Just to check my understanding, you're suggesting it's a spurious warning?

File: src/julia-syntax.scm [hunk 1]
```diff
@@ -3862,7 +3862,7 @@ f(x) = yt(x)
                                            v)))
                                    cvs)))
                `(new_opaque_closure
-                 ,(cadr e) (call (core apply_type) Union) (core Any)
+                 ,(cadr e) (top Bottom) (core Any)
```
Comment (vtjnash) [inline src/julia-syntax.scm:3865]:
```suggestion
                 ,(cadr e) (call (core apply_type) (core Union)) (core Any)
```
(Bottom is not a name used in the system)

Comment (simeonschaub) [reply src/julia-syntax.scm:3865]: I wonder whether we should just define it in Core though. It seems a bit weird to always have to call `apply_type` to get the bottom type in lowering. Seems like replacing it with a single `GlobalRef` would be preferable

Comment (vtjnash) [reply src/julia-syntax.scm:3865]: nah, this is pretty good

**Pull request merged**